osu.Game.Rulesets.Soyokaze/Mods/SoyokazeModHidden.cs

```diff
@@ -65,9 +80,12 @@ private void applyState(DrawableHitObject drawableObject, bool increaseVisibilit
                 case DrawableHitCircle circle:
                     fadeTarget = circle.HitCircle;
                     if (!increaseVisibility)
-                        using (circle.BeginAbsoluteSequence(hitObject.StartTime - hitObject.Preempt))
+                        using (circle.BeginAbsoluteSequence(FadingApproachCircle.Value? fadeOutStartTime : hitObject.StartTime - hitObject.Preempt))
                         {
-                            circle.ApproachCircle.Hide();
+                            if (FadingApproachCircle.Value)
+                                circle.ApproachCircle.FadeOut(fadeOutDuration / FadingApproachCircleSpeed.Value);
```

Comment:
(same as what I said above) For example, you could make this `fadeOutDuration` or `fadeOutDuration * 0.75` or something like that.